sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -308,6 +308,10 @@ case class MapPartitions[T, U](
     output: Seq[Attribute],
     child: SparkPlan) extends UnaryNode {
 
+  override def canProcessSafeRows: Boolean = true
+  override def canProcessUnsafeRows: Boolean = true
+  override def outputsUnsafeRows: Boolean = child.outputsUnsafeRows
```

Comment:
This is not correct right?  It will always output unsafe rows because encoders always output unsafe rows.